javascript/elements/cable_ready_element.js

```diff
@@ -0,0 +1,36 @@
+import { perform } from '../cable_ready'
+
+export default class CableReadyElement extends HTMLElement {
+  static define () {
+    if (!customElements.get('cable-ready')) {
```

Comment:
It seems like either this should be moved to https://github.com/stimulusreflex/cable_ready/blob/9fdbc8996d724473909ab708e1ad4cf25bad5386/javascript/elements/index.js#L28, or all of the other elements should be updated to use this `static define()` form. Which should it be?

Reply:
So from what I understand, that particular function tries to set up CR connections and will throw an error if there's a missing consumer. I'd like the CR element itself to standalone, so a user needs to be able to import it by itself. Now whether we let them handle the `customElements.define` or we do it automatically upon import or we use a `define()` method, certainly up for debate.

Reply:
And in case my reasoning isn't super-clear, I personally would often use the CR element outside of Rails & ActionCable entirely, hence the need for maximum compatibility.

Reply:
I'd love to get @julianrubisch's and @marcoroth's :eyes: on this, but it seems like we should be consistent wrt the `define` technique. I'd be fine with this PR moving the defines for the other two elements into the elements themselves, as you have done for `cable-ready`.

As for the standalone question... I concede that this particular element does not need access to the `CableConsider` object which we rely on for the other two.

Reply:
> So from what I understand, that particular function tries to set up CR connections and will throw an error if there's a missing consumer

@jaredcwhite This function is the initialize function to setup the whole library, not just the consumer. It's usually being call in the app code in `application.js`. So it would make sense to keep the `define` calls here.

But maybe we can move the define logic into the elements itself, so that the elements can handle themselves how they are being declared. This also has the benefit that the name of the element is at the same place as the code for the element itself.. And instead of defining it directly in the `initialize()` function we just call the static `declare()` function on every element.

Something like:
```js
// javascript/elements/index.js

import StreamFromElement from './stream_from_element'
import UpdatesForElement from './updates_for_element'
import CableReadyElement from './cable_ready_element'


// ...

const initialize = (initializeOptions = {}) => {
 // ..

 StreamFromElement.declare()
 UpdatesForElement.declare()
 CableReadyElement.declare()

 // ..
}
```

Reply:
@marcoroth @leastbad OK I wrapped my head around this, so I've updated the PR with the consolidated logic for defining the elements.